### ISSUE TITLE
The confusing behaviour of slices

### ISSUE THREAD

**Issue by greencoloured** [fish-shell/fish-shell#5187]
```
fish, version 2.7.1
Linux h001201355 4.17.19-200.fc28.x86_64 #1 SMP Fri Aug 24 15:47:41 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
xterm-256color
# GNOME Terminal 3.28.2 using VTE 0.52.2 +GNUTLS
```
Tested with customisations disabled.

The behaviour of array slices in fish (and lack of `shift`?) make them confusing. 

```fish
function t; echo "First:" $argv[1] "Rest: $argv[2..-1]"; end
t 1 2 3 
# First: 1 Rest: 2 3
t 1 2
# First: 1 Rest: 2
t 1
# First: 1 Rest: 1
```
It appears that since slices can be low to high, the idiom of saying position "2 to end" will actually repeat the first element of there is only one.

And you have people suggesting top voted stack overflow answers that are broken:
 
https://stackoverflow.com/questions/24093649/how-to-access-remaining-arguments-in-a-fish-script

**Comment by faho** [comment]
This was supposed to be solved in #4965, which it seemingly isn't.

> (and lack of shift?) 

`set -e var[1]` does the same thing.

**Comment by greencoloured** [comment]
> This was supposed to be solved in #4965, which it seemingly isn't.
> 
> > (and lack of shift?)
> 
> `set -e var[1]` does the same thing.

Thanks for the info on the shift replacement. Hope you get the slice behaviour ironed out.
